# Autogen Installation

In [1]:
# Autogen Installation

1. Create a venv
2. Create a requirements.txt
3. install autogen-agentchat autogen-core autogen-ext

# First Autogen Agent

In [6]:
from autogen_agentchat.agents import AssistantAgent
from autogen_ext.models.openai import OpenAIChatCompletionClient
from dotenv import load_dotenv

import os

load_dotenv()

api_key = os.getenv('OPENAI_API_KEY')

# Connecting to the Model

In [8]:
OpenAI_client = OpenAIChatCompletionClient(model="gpt-4o",api_key=api_key)


In [9]:
OpenAI_Agent = AssistantAgent(name="my_assistant",model_client=OpenAI_client)

In [16]:
result = await OpenAI_Agent.run(task="Tell me something about you?")

In [17]:
print(result)

messages=[TextMessage(id='76dab185-66ea-4b99-b274-6925d93e5b48', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 5, 35, 962185, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='b678dcae-c8a2-45f5-999e-934b48b79f62', source='my_assistant', models_usage=RequestUsage(prompt_tokens=203, completion_tokens=88), metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 5, 37, 303623, tzinfo=datetime.timezone.utc), content="I am an AI assistant created by OpenAI, designed to help answer questions and provide information on a wide range of topics. My primary purpose is to assist users by processing text, generating suggestions, and offering relevant insights based on the data I've been trained on. While I don't have direct access to external systems or real-time data, I'm here to support users by offering the knowledge I've been built with up until October 2023. TERMINATE", type='TextMessage')] 

messages=[TextMessage(id='76dab185-66ea-4b99-b274-6925d93e5b48', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 5, 35, 962185, tzinfo=datetime.timezone.utc), content='Tell me something about you?', type='TextMessage'), TextMessage(id='b678dcae-c8a2-45f5-999e-934b48b79f62', source='my_assistant', models_usage=RequestUsage(prompt_tokens=203, completion_tokens=88), metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 5, 37, 303623, tzinfo=datetime.timezone.utc), content="I am an AI assistant created by OpenAI, designed to help answer questions and provide information on a wide range of topics. My primary purpose is to assist users by processing text, generating suggestions, and offering relevant insights based on the data I've been trained on. While I don't have direct access to external systems or real-time data, I'm here to support users by offering the knowledge I've been built with up until October 2023. TERMINATE", type='TextMessage')] stop_reason=None

In [19]:
result2 = await OpenAI_Agent.run(task='What was my last question?')

In [20]:
result2

TaskResult(messages=[TextMessage(id='35014e79-e8f5-407d-bdab-8b21c60415e8', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 6, 22, 906859, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='689bb3da-40ec-493c-b89d-4609b3b9918d', source='my_assistant', models_usage=RequestUsage(prompt_tokens=306, completion_tokens=14), metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 6, 23, 882783, tzinfo=datetime.timezone.utc), content='Your last question was: "Tell me something about you?" TERMINATE', type='TextMessage')], stop_reason=None)

TaskResult(messages=[TextMessage(id='35014e79-e8f5-407d-bdab-8b21c60415e8', source='user', models_usage=None, metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 6, 22, 906859, tzinfo=datetime.timezone.utc), content='What was my last question?', type='TextMessage'), TextMessage(id='689bb3da-40ec-493c-b89d-4609b3b9918d', source='my_assistant', models_usage=RequestUsage(prompt_tokens=306, completion_tokens=14), metadata={}, created_at=datetime.datetime(2025, 7, 5, 7, 6, 23, 882783, tzinfo=datetime.timezone.utc), content='Your last question was: "Tell me something about you?" TERMINATE', type='TextMessage')], stop_reason=None)